hal/inc/hal_dynalib_spi.h

```diff
@@ -55,6 +55,8 @@ DYNALIB_FN(12, hal_spi, HAL_SPI_Set_Callback_On_Select, void(HAL_SPI_Interface,
 DYNALIB_FN(13, hal_spi, HAL_SPI_DMA_Transfer_Cancel, void(HAL_SPI_Interface))
 DYNALIB_FN(14, hal_spi, HAL_SPI_DMA_Transfer_Status, int32_t(HAL_SPI_Interface, HAL_SPI_TransferStatus*))
 DYNALIB_FN(15, hal_spi, HAL_SPI_Set_Settings, int32_t(HAL_SPI_Interface, uint8_t, uint8_t, uint8_t, uint8_t, void*))
+DYNALIB_FN(16, hal_spi, HAL_SPI_Acquire, int32_t(HAL_SPI_Interface, void*))
```

Comment:
These are only available on Gen 3 platforms, so let's introduce a feature flag to `hal_platform.h` (e.g. `HAL_PLATFORM_SPI_HAL_THREAD_SAFETY`) and conditionally export these functions for Gen 3 only.